### DISCUSSION TITLE
OAuth 2.0 Token Exchange (RFC8693) is now fully supported (it's not?)

### DISCUSSION THREAD

**Discussion by denysandriyanov** [ory/hydra#3359] [Q&A] [answered]
Hello, in release notes for Hydra v2 it's stated that RFC8693 is fully supported, while when I try to post according to the the flow i am getting 

	/go/pkg/mod/github.com/prometheus/client_golang@v1.11.0/prometheus/promhttp/instrument_server.go:68
net/http.HandlerFunc.ServeHTTP
	/usr/local/go/src/net/http/server.go:2109
github.com/prometheus/client_golang/prometheus/promhttp.InstrumentHandlerDuration.func2
	/go/pkg/mod/github.com/prometheus/client_golang@v1.11.0/prometheus/promhttp/instrument_server.go:76
net/http.HandlerFunc.ServeHTTP
	/usr/local/go/src/net/http/server.go:2109
github.com/prometheus/client_golang/prometheus/promhttp.InstrumentHandlerRequestSize.func1
	/go/pkg/mod/github.com/prometheus/client_golang@v1.11.0/prometheus/promhttp/instrument_server.go:165
net/http.HandlerFunc.ServeHTTP
	/usr/local/go/src/net/http/server.go:2109
github.com/ory/x/prometheusx.Metrics.instrumentHandlerStatusBucket.func1
	/go/pkg/mod/github.com/ory/x@v0.0.486/prometheusx/metrics.go:108
net/http.HandlerFunc.ServeHTTP
	/usr/local/go/src/net/http/server.go:2109
github.com/ory/x/prometheusx.(*MetricsManager).ServeHTTP
	/go/pkg/mod/github.com/ory/x@v0.0.486/prometheusx/middleware.go:36
github.com/urfave/negroni.middleware.ServeHTTP
	/go/pkg/mod/github.com/urfave/negroni@v1.0.0/negroni.go:38 status:Bad Request status_code:400]

**Comment by vinckr** [comment]
Hello @denysandriyanov 
Could you please share a reproducible repository with shell commands (e.g. hydra clients create, curl ...) that reproduces this issue?

**Comment by aeneasr** [comment, marked as answer]
The release notes here are incorrect - sorry about that. We have since (I believe) fixed the release notes!

**Reply by nryissuu** [reply]
I'm also following [https://www.ory.sh/docs/hydra/guides/jwt](https://www.ory.sh/docs/hydra/guides/jwt) and I get the same error.
Wanted to use JWTs internally but keep getting an error

{`"error":"invalid_request","error_description":"The request is missing a required parameter, includes an invalid parameter value, includes a parameter more than once, or is otherwise malformed. Client credentials missing or malformed in both HTTP Authorization header and HTTP POST body."}`

Are JWTs not supported yet?

Also _token_endpoint_auth_method_ seems to only allow _client_secret_basic_.
Am I missing something?

Thank you

**Comment by pferrot** [comment]
You should fix this page: https://www.ory.sh/oauth2-server-openid-ory-hydra-v2/

![image](https://github.com/ory/hydra/assets/12295384/7ab3a511-0ea0-463c-b863-8fd6b3b96b48)

This is misleading and one can easily stumble upon it with a simple Google search.

![image](https://github.com/ory/hydra/assets/12295384/f0927a96-a9af-48dc-8a2f-8fa6f0c71b3f)

**Reply by vinckr** [reply]
Thanks @pferrot 
We have fixed it in the official release notes but missed this blogpost, fix is in pipeline.
